src/java/org/apache/cassandra/metrics/ClientMetrics.java

```diff
@@ -136,16 +140,19 @@ private int countConnectedClients()
 
         for (Server server : servers)
             for (ClientStat stat : server.recentClientStats())
-                stats.add(new HashMap(stat.asMap())); // asMap returns guava, so need to convert to java for jmx
+                stats.add(new HashMap<>(stat.asMap())); // asMap returns guava, so need to convert to java for jmx
 
         stats.sort(Comparator.comparing(map -> map.get(ClientStat.PROTOCOL_VERSION)));
 
         return stats;
     }
 
-    private <T> Gauge<T> registerGauge(String name, Gauge<T> gauge)
+    private <T> Gauge<T> registerGauge(String name, String deprecated, Gauge<T> gauge)
```

Comment:
Thinking this would be cleaner to have two methods

```
private <T> Gauge<T> registerGauge(String name, String deprecated, Gauge<T> gauge)
{
    return Metrics.register(factory.createMetricName(name), gauge, factory.createMetricName(deprecated));
}

private <T> Gauge<T> registerGauge(String name, Gauge<T> gauge)
{
    return Metrics.register(factory.createMetricName(name), gauge);
}
```

current logic is mostly rewriting that, if one method is desired, would be best to create the `MetricName... aliases` in this method then.

Reply:
Did something similar, but in the direction of avoiding having the same metric name creation code in both methods.